sdk/nodejs/output.ts

```diff
@@ -353,14 +385,18 @@ export function output<T>(val: Input<T | undefined>): Output<Unwrap<T | undefine
         // Outputs and we want to preserve them as is when flattening.
         return createSimpleOutput(val);
     }
+    else if (isUnknown(val)) {
+        // Turn unknowns into unknown outputs.
+        return <any>new Output(new Set(), Promise.resolve(<any>val), /*isKnown*/ Promise.resolve(false), /*isSecret*/ Promise.resolve(false));
+    }
     else if (val instanceof Promise) {
         // For a promise, we can just treat the same as an output that points to that resource. So
         // we just create an Output around the Promise, and immediately apply the unwrap function on
         // it to transform the value it points at.
-        return <any>new Output(new Set(), val, /*isKnown*/ Promise.resolve(true), /*isSecret*/ Promise.resolve(false)).apply(output);
+        return <any>(<any>new Output(new Set(), val, /*isKnown*/ Promise.resolve(true), /*isSecret*/ Promise.resolve(false))).apply(output, true);
     }
     else if (Output.isInstance(val)) {
-        return <any>val.apply(output);
+        return <any>(<any>val).apply(output, /*runWithUnknowns:*/ true);
```

Comment:
Why are these used here?  If observable - should the doc comment on the function be updated?

Reply:
This is necessary s.t. inner unknowns are properly propagated up to the top level output.